core/common/boot/src/main/java/org/eclipse/edc/boot/system/DefaultServiceExtensionContext.java

```diff
@@ -108,20 +113,21 @@ public void initialize() {
             getMonitor().warning("The runtime is configured as an anonymous participant. DO NOT DO THIS IN PRODUCTION.");
         }
 
-        var connectorName = getSetting(EDC_CONNECTOR_NAME, null);
-        if (connectorName != null) {
-            getMonitor().warning("Setting %s has been deprecated, please use %s instead".formatted(EDC_CONNECTOR_NAME, RUNTIME_ID));
+        runtimeId = getSetting(RUNTIME_ID, null);
+        if (runtimeId != null) {
+            getMonitor().warning("A configuration value for '%s' was found. The runtime ID should be left blank to use a random one.".formatted(RUNTIME_ID));
         }
 
-        runtimeId = getSetting(RUNTIME_ID, null);
-        if (runtimeId == null) {
-            if (connectorName == null) {
-                getMonitor().warning("%s is not configured so a random UUID is used. It is recommended to provide a static one.".formatted(RUNTIME_ID));
-                runtimeId = UUID.randomUUID().toString();
+        componentId = getSetting(COMPONENT_ID, null);
+        if (componentId == null) {
+            if (runtimeId != null) {
+                getMonitor().warning("'%s' is not configured, but '%s' is. The value used for '%s' will be used for '%s', and a random string will be used for '%s'".formatted(COMPONENT_ID, RUNTIME_ID, RUNTIME_ID, COMPONENT_ID, RUNTIME_ID));
+                componentId = runtimeId;
             } else {
-                getMonitor().warning("%s is not configured and it will fallback to the deprecated %s value".formatted(RUNTIME_ID, EDC_CONNECTOR_NAME));
-                runtimeId = connectorName;
+                getMonitor().warning("%s is not configured so a random UUID is used. It is recommended to provide a static one.".formatted(COMPONENT_ID));
+                componentId = UUID.randomUUID().toString();
```

Comment:
this logic could be simplified, what about:
`runtime.id`:
- if `runtime.id` is not null print warning
- else set random `runtime.id`

`component.id`
- if `component.id` is null print warning and set it equal to `runtime.id` (that will be randomic if `runtime.id` was null otherwise will be the same value as `runtime.id`)

Reply:
that is not quite the same though, because I wanted `runtime.id` to be _always_ randomized.

Reply:
ok, gotcha, I guess in a future we could just avoid reading the `edc.runtime.id` setting at all.

Reply:
yes, this is for backwards compatibility mostly